packages/flutter/lib/src/material/reorderable_list.dart

```diff
@@ -596,3 +596,30 @@ class _ReorderableListContentState extends State<_ReorderableListContent> with T
     });
   }
 }
+
+/// A global key that takes its identity from the object and uses a value of a
+/// particular type to identify itself.
+///
+/// The difference with GlobalObjectKey is that it uses [==] instead of [identical]
+/// of the objects used to generate widgets.
+@optionalTypeArgs
+class _ReorderableListViewChildGlobalKey extends GlobalObjectKey {
+
+  const _ReorderableListViewChildGlobalKey(this.subKey, this.state) : super(subKey);
+
+  final Key subKey;
+
+  final _ReorderableListContentState state;
+
+  @override
+  bool operator ==(Object other) {
+    if (other.runtimeType != runtimeType)
+      return false;
+    return other is _ReorderableListViewChildGlobalKey
+        && other.subKey == subKey
+        && other.state == state;
+  }
+
+  @override
+  int get hashCode => identityHashCode(subKey);
```

Comment:
```
    If [operator ==] is overridden to use the object state instead,
    the hash code must also be changed to represent that state.
```

Reply:
How about this way?

```Dart
@override
int get hashCode => hashValues(subKey.hashCode, identityHashCode(state));
```

Reply:
The new hash seems to work fine :)

Reply:
```dart
@override
int get hashCode => hashValues(subKey, state, index);
```

Reply:
Do we need an index? ValueKey already has a value for the index.

Reply:
See https://github.com/flutter/flutter/pull/64855#discussion_r484312001